### PR TITLE
Track ongoing search tasks

### PR DESCRIPTION
**Pull request by JVerwolf**
Tracks the ongoing tasks in the Search and Search Coodinator executors.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java [hunk 1]
```diff
@@ -222,15 +222,15 @@ public ThreadPool(final Settings settings, MeterRegistry meterRegistry, final Ex
         builders.put(Names.ANALYZE, new FixedExecutorBuilder(settings, Names.ANALYZE, 1, 16, TaskTrackingConfig.DO_NOT_TRACK));
         builders.put(
             Names.SEARCH,
-            new FixedExecutorBuilder(settings, Names.SEARCH, searchOrGetThreadPoolSize, 1000, TaskTrackingConfig.DEFAULT)
+            new FixedExecutorBuilder(settings, Names.SEARCH, searchOrGetThreadPoolSize, 1000, new TaskTrackingConfig(true, 0.1))
```
Comment (JVerwolf) [inline server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java:225]: The default EWMA (exponential weighted moving average) alpha value is `0.3`. I've updated this to `0.1` be remain consistent with the `WRITE`,`SYSTEM_WRITE`, and `SYSTEM_CRITICAL_WRITE` threadpools which are used for index-load autoscaling.

**Comment by JVerwolf** [comment]
buildkite test this

**Review by original-brownbear — APPROVED** [review]
LGTM, but maybe do the constant refactoring and I think I'd make this a non-issue, no need for it to appear separately in the release log.

File: server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java [hunk 2]
```diff
@@ -151,6 +151,8 @@ public static ThreadPoolType fromType(String type) {
         entry(Names.SYSTEM_CRITICAL_WRITE, ThreadPoolType.FIXED)
     );
 
+    public static final double searchAutoscalingEWMA = 0.1;
```
Comment (JVerwolf) [inline server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java:154]: See https://github.com/elastic/elasticsearch-serverless/pull/445 for background on this number.

**Pull request merged**